### PR TITLE
MODSIDECAR-20: configure TLS Web client 

### PR DESCRIPTION
**Pull request by yaroslav-kiriak**
## Purpose
Configure TLS Web client  to call manager modules, egress requests and users

## TODOS and Open Questions

- [ ] Check logging

## Pre-Merge Checklist:

Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
    - [ ] Code coverage on new code is 80% or greater
    - [ ] Duplications on new code is 3% or less
    - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
    - [ ] Were any API paths or methods changed, added, or removed?
    - [ ] Were there any schema changes?
    - [ ] Did any of the interface versions change?
    - [ ] Were permissions changed, added, or removed?
    - [ ] Are there new interface dependencies?
    - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do Rally stories exist to update the impacted modules?
    - [ ] If not, please create them
    - [ ] Do they contain the appropriate level of detail? Which endpoints/schemas changed, etc.
    - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the Rally stories under active development?
    - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
    - [ ] If so, have they been approved?

Ideally, all the PRs involved in breaking changes would be merged on the same day to avoid breaking the folio-testing
environment. Communication is paramount if that is to be achieved, especially as the number of inter-module and
inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the
responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/sidecar/integration/users/UserService.java [hunk 1]
```diff
@@ -12,22 +12,29 @@
...
-@RequiredArgsConstructor
 public class UserService {
 
-  private final WebClient webClient;
+  private final WebClient webClientTls;
   private final ModUsersProperties modUsersProperties;
   private final Cache<String, User> userCache;
   private final ServiceTokenProvider serviceTokenProvider;
 
+  public UserService(@Named("webClientTls") WebClient webClientTls, ModUsersProperties modUsersProperties,
```
Comment (pfilippov-epam) [inline src/main/java/org/folio/sidecar/integration/users/UserService.java:30]: you can add `@Named` annotation to the `lombok.config` and continue using `@RequiredArgsConstructor`

File: src/main/java/org/folio/sidecar/service/routing/EgressRequestHandler.java [hunk 2]
```diff
@@ -122,10 +122,10 @@ private boolean requireSystemUserToken(RoutingContext rc) {
   }
 
   private void forwardRequest(RoutingContext rc, HttpServerRequest rq, String moduleId, String url,
-    String updatedPath) {
+                              String updatedPath) {
```
Comment (pfilippov-epam) [inline src/main/java/org/folio/sidecar/service/routing/EgressRequestHandler.java:125]: This change is not required, looks like that you don't add idea configuration from code uniformity page

File: src/main/java/org/folio/sidecar/integration/tm/TenantManagerClient.java [hunk 3]
```diff
@@ -23,13 +23,19 @@
...
-@RequiredArgsConstructor
 public class TenantManagerClient {
 
-  private final WebClient webClient;
+  private final WebClient webClientTls;
   private final JsonConverter jsonConverter;
   private final TenantManagerClientProperties clientProperties;
 
+  public TenantManagerClient(@Named("webClientTls") WebClient webClientTls, JsonConverter jsonConverter,
+                             TenantManagerClientProperties clientProperties) {
```
Comment (pfilippov-epam) [inline src/main/java/org/folio/sidecar/integration/tm/TenantManagerClient.java:33]: you can add `@Named` annotation to the `lombok.config` and continue using `@RequiredArgsConstructor`

File: src/main/java/org/folio/sidecar/integration/te/TenantEntitlementClient.java [hunk 4]
```diff
@@ -19,14 +20,14 @@
...
 
-  private final WebClient webClient;
+  private final WebClient webClientTls;
   private final JsonConverter jsonConverter;
   private final TenantEntitlementClientProperties clientProperties;
 
-  public TenantEntitlementClient(WebClient webClient, JsonConverter jsonConverter,
-    TenantEntitlementClientProperties clientProperties) {
+  public TenantEntitlementClient(@Named("webClientTls") WebClient webClientTls, JsonConverter jsonConverter,
+                                 TenantEntitlementClientProperties clientProperties) {
```
Comment (pfilippov-epam) [inline src/main/java/org/folio/sidecar/integration/te/TenantEntitlementClient.java:28]: Same, this intendation is not correct

File: src/main/java/org/folio/sidecar/integration/te/TenantEntitlementClient.java [hunk 5]
```diff
@@ -19,14 +20,14 @@
 @ApplicationScoped
 public class TenantEntitlementClient {
 
-  private final WebClient webClient;
+  private final WebClient webClientTls;
```
Comment (pfilippov-epam) [inline src/main/java/org/folio/sidecar/integration/te/TenantEntitlementClient.java:23]: There is no need to rename webClient

File: src/main/java/org/folio/sidecar/integration/keycloak/KeycloakClient.java [hunk 6]
```diff
@@ -11,20 +11,24 @@
...
 
 @Log4j2
 @ApplicationScoped
-@RequiredArgsConstructor
 public class KeycloakClient {
 
   private final WebClient webClient;
   private final KeycloakProperties properties;
 
+  public KeycloakClient(@Named("webClient") WebClient webClient, KeycloakProperties properties) {
```
Comment (pfilippov-epam) [inline src/main/java/org/folio/sidecar/integration/keycloak/KeycloakClient.java:27]: you can add `@Named` annotation to the `lombok.config` and continue using `@RequiredArgsConstructor`

File: src/main/java/org/folio/sidecar/integration/am/ApplicationManagerClient.java [hunk 7]
```diff
@@ -7,20 +7,26 @@
...
 import org.folio.sidecar.integration.am.model.ModuleBootstrap;
 import org.folio.sidecar.service.JsonConverter;
 
 @Log4j2
 @ApplicationScoped
-@RequiredArgsConstructor
 public class ApplicationManagerClient {
 
-  private final WebClient webClient;
+  private final WebClient webClientTls;
```
Comment (pfilippov-epam) [inline src/main/java/org/folio/sidecar/integration/am/ApplicationManagerClient.java:19]: There is no need to rename a webClient object

File: src/main/java/org/folio/sidecar/integration/am/ApplicationManagerClient.java [hunk 8]
```diff
@@ -7,20 +7,26 @@
...
   private final JsonConverter jsonConverter;
   private final AppManagerClientProperties clientProperties;
 
+  public ApplicationManagerClient(@Named("webClientTls") WebClient webClientTls, JsonConverter jsonConverter,
+                                  AppManagerClientProperties clientProperties) {
```
Comment (pfilippov-epam) [inline src/main/java/org/folio/sidecar/integration/am/ApplicationManagerClient.java:24]: This intendation is not correct

File: src/main/java/org/folio/sidecar/integration/tm/TenantManagerClient.java [hunk 9]
```diff
@@ -23,13 +23,19 @@
 
 @Log4j2
 @ApplicationScoped
-@RequiredArgsConstructor
 public class TenantManagerClient {
 
-  private final WebClient webClient;
+  private final WebClient webClientTls;
```
Comment (pfilippov-epam) [inline src/main/java/org/folio/sidecar/integration/tm/TenantManagerClient.java:28]: There is no need to rename a webClient object

File: src/main/java/org/folio/sidecar/service/routing/RequestForwardingService.java [hunk 10]
```diff
@@ -37,10 +37,22 @@ public class RequestForwardingService {
   private static final Predicate<String> HEADERS_PREDICATE = header -> !USER_AGENT.equalsIgnoreCase(header);
 
   private final WebClient webClient;
+  private final WebClient webClientTls;
```
Comment (pfilippov-epam) [inline src/main/java/org/folio/sidecar/service/routing/RequestForwardingService.java:40]: Same as above, no need to rename

File: src/main/java/org/folio/sidecar/service/routing/RequestForwardingService.java [hunk 11]
```diff
@@ -37,10 +37,22 @@ public class RequestForwardingService {
...
   private final ErrorHandler errorHandler;
   private final SidecarSignatureService sidecarSignatureService;
   private final WebClientProperties webClientProperties;
 
+  public RequestForwardingService(@Named("webClient") WebClient webClient,
+                                  @Named("webClientTls") WebClient webClientTls,
+                                  ErrorHandler errorHandler, SidecarSignatureService sidecarSignatureService,
+                                  WebClientProperties webClientProperties) {
```
Comment (pfilippov-epam) [inline src/main/java/org/folio/sidecar/service/routing/RequestForwardingService.java:48]: Indentation is not correct

File: src/main/java/org/folio/sidecar/service/routing/RequestForwardingService.java [hunk 12]
```diff
@@ -49,13 +61,16 @@ public class RequestForwardingService {
    */
   @SneakyThrows
   public void forward(RoutingContext rc, String absUri) {
+    forward(rc, absUri, webClient);
+  }
+
+  private void forward(RoutingContext rc, String absUri, WebClient webClient) {
```
Comment (pfilippov-epam) [inline src/main/java/org/folio/sidecar/service/routing/RequestForwardingService.java:67]: Private method is above on public method

File: src/test/java/org/folio/sidecar/integration/am/ApplicationManagerClientTest.java [hunk 13]
```diff
@@ -41,7 +41,7 @@ class ApplicationManagerClientTest {
 
   @Spy private final JsonConverter jsonConverter = new JsonConverter(OBJECT_MAPPER);
 
-  @Mock private WebClient webClient;
+  @Mock private WebClient webClientTls;
```
Comment (pfilippov-epam) [inline src/test/java/org/folio/sidecar/integration/am/ApplicationManagerClientTest.java:44]: Same as above, no need to rename

File: src/test/java/org/folio/sidecar/integration/te/TenantEntitlementClientTest.java [hunk 14]
```diff
@@ -43,7 +43,7 @@ class TenantEntitlementClientTest {
 
   private final JsonConverter jsonConverter = new JsonConverter(OBJECT_MAPPER);
 
-  @Mock private WebClient webClient;
+  @Mock private WebClient webClientTls;
```
Comment (pfilippov-epam) [inline src/test/java/org/folio/sidecar/integration/te/TenantEntitlementClientTest.java:46]: Same as above, no need to rename

File: src/test/java/org/folio/sidecar/integration/tm/TenantManagerClientTest.java [hunk 15]
```diff
@@ -43,7 +43,7 @@ class TenantManagerClientTest {
 
   private final JsonConverter jsonConverter = new JsonConverter(OBJECT_MAPPER);
 
-  @Mock private WebClient webClient;
+  @Mock private WebClient webClientTls;
```
Comment (pfilippov-epam) [inline src/test/java/org/folio/sidecar/integration/tm/TenantManagerClientTest.java:46]: Same as above, no need to rename

File: src/test/java/org/folio/sidecar/integration/users/UserServiceTest.java [hunk 16]
```diff
@@ -36,7 +36,7 @@ class UserServiceTest {
 
   @InjectMocks private UserService userService;
 
-  @Mock private WebClient webClient;
+  @Mock private WebClient webClientTls;
```
Comment (pfilippov-epam) [inline src/test/java/org/folio/sidecar/integration/users/UserServiceTest.java:39]: Same as above, no need to rename

**Review by pfilippov-epam — CHANGES_REQUESTED** [review]

File: src/main/java/org/folio/sidecar/service/routing/RequestForwardingService.java (hunk 12 above)
Comment (yaroslav-kiriak) [reply src/main/java/org/folio/sidecar/service/routing/RequestForwardingService.java:67]: Right, I just had a checkstyle violation. I will suppress it then

File: src/main/java/org/folio/sidecar/integration/users/UserService.java (hunk 1 above)
Comment (pfilippov-epam) [reply src/main/java/org/folio/sidecar/integration/users/UserService.java:30]: > you can add `@Named` annotation to the `lombok.config` and continue using `@RequiredArgsConstructor`

No need to do that, Intellij idea provides a warning for this case:
`Private access modifier usage is not recommended for Quarkus CDI members`

**Review by pfilippov-epam — APPROVED** [review]

**Review by dmtkachenko — APPROVED** [review]

**Pull request merged**